### PR TITLE
56279 - pos batch file attributes changes

### PR DESCRIPTION
**Pull request by Vishal14583**
Remove Batch Type from attributes for all batches.
Remvoe S63 version for catalogue and ENC Update batch.
Add file details like MimeType, FileSize in stubs etc.
Add file name in file attribute list for all batches.

### CODE REVIEW

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/FssService.cs [hunk 1]
```diff
@@ -23,18 +24,21 @@ public class FssService : IFssService
...
-                               IFileSystemHelper fileSystemHelper)
+                               IFileSystemHelper fileSystemHelper,
+                               IConfiguration configuration)
         {
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
             _fssApiConfiguration = fssApiConfiguration ?? throw new ArgumentNullException(nameof(fssApiConfiguration));
             _fssApiClient = fssApiClient ?? throw new ArgumentNullException(nameof(fssApiClient));
             _authFssTokenProvider = authFssTokenProvider ?? throw new ArgumentNullException(nameof(authFssTokenProvider));
             _fileSystemHelper = fileSystemHelper ?? throw new ArgumentNullException(nameof(fileSystemHelper));
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration)); ;
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/FssService.cs:41]:
```suggestion
            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
```

**Review by harshal11869 — CHANGES_REQUESTED** [review]

**Review by mattwin1 — CHANGES_REQUESTED** [review]

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Helpers/FileHelper.cs [hunk 2]
```diff
@@ -63,5 +64,13 @@ public static bool CleanUp(string filePath)
             }
             return false;
         }
+
+        public static string GetFileMD5(FileInfo fileInfo)
+        {
+            using Stream? fs = fileInfo.OpenRead();
```
[the remark this answers is not in the document]
Comment (sumod13943) [reply UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Helpers/FileHelper.cs]: Changed

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/GetBatchDetails.cs [hunk 3]
```diff
@@ -34,17 +34,26 @@ public static void GetBatchDetailsResponseValidation(dynamic batchDetailsRespons
...
+
+            string hash = batchDetailsResponse.files[0].hash;
+            hash.Should().NotBeNullOrEmpty();
         }
 
         public static void GetBatchDetailsResponseValidationForFullAVCSExchangeSet(dynamic batchDetailsResponse)
         {
-            string mediaType = batchDetailsResponse.attributes[6].value;
+            string mediaType = batchDetailsResponse.attributes[5].value;
             if (mediaType.Equals("Zip"))
```
[the remark this answers is not in the document]
Comment (sumod13943) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/GetBatchDetails.cs]: Changed

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/FssService.cs (hunk 1 above)
Comment (sumod13943) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Fulfilment/Services/FssService.cs]: Changed

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/FileContentHelper.cs [hunk 4]
```diff
@@ -41,7 +41,7 @@ public static async Task<List<string>> DownloadAndExtractExchangeSetZipFileForLa
         {
             List<string> downloadFolderPath = new();
 
-            string mediaType = batchDetailsResponse.attributes[6].value;
+            string mediaType = batchDetailsResponse.attributes[5].value;
 
             if (mediaType.Equals("Zip"))
```
[the remark this answers is not in the document]
Comment (sumod13943) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/FileContentHelper.cs]: Changed

**Review by harshal11869 — APPROVED** [review]
Looks Good

File: azure-pipelines.yml [hunk 5]
```diff
@@ -496,7 +496,7 @@ stages:
                     scriptType: 'pscore'
                     scriptLocation: 'scriptPath'
                     scriptPath: "$(Build.SourcesDirectory)/devterraformartifact/set_stub_webjob_configuration.ps1"
-                    arguments: '-essapibaseurl $(ESSApiConfiguration.BaseUrl_PostFT) -fssapibaseurl $(FSSApiConfiguration.BaseUrl_PostFT) -fleetmanagerbaseurl $(FleetManagerB2BApiConfiguration.BaseUrl_PostFT) -fleetmanagerfilepath $(FleetManagerB2BApiConfiguration.GetCatalogueResponseFilePath_PostFT) -resourcegroup $(RGName) -webappname $(WEB_APP_NAME) -mockresourcegroup $(mockWebAppResourceGroup) -mockwebappname $(mockWebAppName)'
+                    arguments: '-essapibaseurl $(ESSApiConfiguration.BaseUrl_PostFT) -fssapibaseurl $(FSSApiConfiguration.BaseUrl_PostFT) -fleetmanagerbaseurl $(FleetManagerB2BApiConfiguration.BaseUrl_PostFT) -fleetmanagerfilepath $(FleetManagerB2BApiConfiguration.GetCatalogueResponseFilePath_PostFT) -ftrunning $(IsFTRunning_PostFT) -resourcegroup $(RGName) -webappname $(WEB_APP_NAME) -mockresourcegroup $(mockWebAppResourceGroup) -mockwebappname $(mockWebAppName)'
```
[the remark this answers is not in the document]
Comment (surendrabehara2022) [reply azure-pipelines.yml]: Hi Matt,
I have hardcoded the values "true" and "false" instead of referring from library variables.

**Review by ajasonca — APPROVED** [review]
Looks good

**Review by mattwin1 — APPROVED** [review]
LGTM

**Review by ajasonca — APPROVED** [review]

**Pull request merged**